controllers/datafoundationclient.go

```diff
@@ -141,19 +140,11 @@ func (r *dataFoundationClientReconciler) reconcileStorageClient() error {
 		return fmt.Errorf("invalid provider endpoint, empty string")
 	}
 
-	providerEndpoint, err := url.ParseRequestURI(r.dataFoundationClientSpec.providerEndpoint)
```

Comment:
- at a minimum it's better to check the input isn't empty

Reply:
We are checking it.
https://github.com/red-hat-storage/managed-fusion-agent/pull/53/files#diff-82e77c947fe176519786a3f7087431a7642e152d10391aa74b1bbbf7e21cb406R139-R141

Reply:
My bad.